src/providers/export/azureCustomVision.ts

```diff
@@ -15,62 +25,201 @@ export interface IAzureCustomVisionOptions {
     domainId?: string;
 }
 
+export interface ITagList {
+    [index: string]: IAzureCustomVisionTag;
+}
+
 export enum NewOrExisting {
     New = "New Project",
     Existing = "Existing Project",
 }
 
-export class AzureCustomVisionProvider extends ExportProvider<IAzureCustomVisionOptions> {
-    constructor(project: IProject, options: IAzureCustomVisionOptions) {
+/**
+ * @name - Azure Custom Vision Provider
+ * @description - Exports a VoTT project into an Azure custom vision project
+ */
+export class AzureCustomVisionProvider extends ExportProvider<IAzureCustomVisionExportOptions> {
+    private customVisionService: AzureCustomVisionService;
+    private assetService: AssetService;
+
+    constructor(project: IProject, options: IAzureCustomVisionExportOptions) {
         super(project, options);
         Guard.null(options);
+
+        const cusomVisionServiceOptions: IAzureCustomVisionServiceOptions = {
+            apiKey: options.apiKey,
+            baseUrl: "https://southcentralus.api.cognitive.microsoft.com/customvision/v2.2/Training",
+        };
+        this.customVisionService = new AzureCustomVisionService(cusomVisionServiceOptions);
+        this.assetService = new AssetService(this.project);
     }
 
-    public export(): Promise<void> {
-        throw new Error("Method not implemented.");
+    /**
+     * Exports the configured assets to the Azure Custom Vision service
+     * @returns The upload results
+     */
+    public async export(): Promise<IExportResults> {
+        const customVisionTags = await this.syncTags();
+        const assetsToExport = await this.getAssetsForExport();
+        const tagMap = _.keyBy(customVisionTags, "name");
+
+        const createImageTasks = assetsToExport.map((asset) => {
+            return this.uploadAsset(asset, tagMap)
+                .then(() => {
```

Comment:
could we use `await` here too instead of `then`?

Reply:
No - this map is creating an array of tasks which are then processed in parallel below... The then/catch here is wrapping the response into a result object since some items can fail.  This allows the Promise.all() to always wait and return a resolved promise with the full results.